static/js/settings_realm_user_settings_defaults.js

```diff
@@ -26,12 +26,16 @@ export function update_page(property) {
         return;
     }
     const container = $(realm_default_settings_panel.container);
-    const value = realm_user_settings_defaults[property];
+    let value = realm_user_settings_defaults[property];
 
     if (property === "emojiset") {
         container.find(`input[value=${CSS.escape(value)}]`).prop("checked", true);
         return;
     }
+    if (property === "twenty_four_hour_time") {
+        value = value.toString();
+    }
```

Comment:
This probably deserves a comment for why it's special, basically move the commit message explanation here.  (We can do one for `emojiset` too).  That'll also help make clear that generally the common code should Just Work.